### DISCUSSION TITLE
Minimum Redis ACL permissions for rate-limiting plugin?

### DISCUSSION THREAD

**Discussion by FelixHarvey** [Kong/kong#12921] [Help] [answered]
Hi 👋 

I'm using Redis with the rate limiting plugin on Kong Gateway. I've got a specific Redis user for Kong, and I want to give it the least permissions possible, using an ACL.

For example: `on ~ratelimit:* -@all +@read +@write +@scripting`. 

Are the minimum permissions required documented somewhere? Or does someone with more in depth knowledge of how the plugin works know what this could be?

**Comment by bungle** [comment, marked as answer]
The plugin calls:
- AUTH
- SELECT
- EVAL

And in EVAL it calls:
- EXISTS
- INCR
- INCRBY
- EXPIRE
- EXPIREAT
